docker-compose.yml

```
@@ -0,0 +1,28 @@
+version: '3.3'
+
+services:
+  db:
+    image: postgres:10.1-alpine
+    restart: always
+    environment:
+      POSTGRES_PASSWORD: vespene!
+      POSTGRES_USER: vespene
+      POSTGRES_DB: vespene
+    volumes:
+      - postgres_data:/var/lib/postgresql/data/
+  web:
+    build: .
+    command: python /vespene/manage.py migrate --noinput
+    command: python /vespene/manage.py generate_secret
+    command: python /vespene/manage.py runserver 0.0.0.0:8000
```

I'm not entirely sure (and definitely not comfortable) if we should be daisy-chaining commands like this instead of using an entrypoint.

---

this is perfectly normal and I suggest you read up on what this is doing if you are uncomfortable on this, it is clearly just running run commands inside the container to setup things for Django...

---

@mpdehaan I used your setup instructions basically to get it going using these run commands, in the instructions I have the user setup the superuser by running it via the docker-compose run command because it is interactive and they have to supply information.

Thanks

---

I still need this to call the setup scripts where possible.  It can skip the superuser setup one.

---

On other projects a used a sidekick container with flyway to do the migrations. It requires to rename the db scripts accordingly but it handles the migrations beautifully.

---

I'm giving your compose file a try. As @chopraaa mentioned, not sure how the multiple commands is working for you. The default behaviour seems to be is hiding all but the last value. 

To make it run i added another dependency on an ephemeral container that runs migrations
`version: '3.3'

services:
  db:
    image: postgres:10.1-alpine
    restart: always
    environment:
      POSTGRES_PASSWORD: vespene!
      POSTGRES_USER: vespene
      POSTGRES_DB: vespene
    expose:
      - "5432/tcp"
    volumes:
      - postgres_data:/var/lib/postgresql/data/
  migrations:
    build: .
    environment:
      DATABASE_HOST: 'http://db'
      DATABASE_PORT: '5432'
    command: python manage.py migrate --noinput
    volumes:
      - .:/vespene
    depends_on:
      - db
  web:
    build: .
    environment:
      DATABASE_HOST: 'http://db'
      DATABASE_PORT: '5432'
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - .:/vespene
    ports:
      - 8000:8000
    depends_on:
      - db
      - migrations`